changes/pr4971.yaml

```
@@ -0,0 +1,2 @@
+fix:
+  - "Allow max_retries to be set to 0 when overriding the config level parameter - [#4971](https://github.com/PrefectHQ/prefect/pull/4971)"
```

```suggestion
  - "Allow disabling retries for a task with `max_retries=0` when retries are globally configured - [#4971](https://github.com/PrefectHQ/prefect/pull/4971)"
```
I try to make these messages address user stories since they end up in the public changelog